### PR TITLE
removed a couple default values from resources properties 

### PR DESCRIPTION
**Pull request by Stromweld**
### Description

Removed a couple of default values from resource properties and added reload example to haproxy_service section of readme. Default values were removed so if they are defined in the default section of the haproxy.cfg file they can be inherited automatically by listeners, frontends, etc... With property specifying a default then it would put that config piece in the listener even if you intended it to be inherited from the default config section.

### Contribution Check List

- [x] All tests pass.
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable

### CODE REVIEW

**Comment by Stromweld** [comment]
ok this should be complete now and ready for review.

File: templates/default/haproxy.cfg.erb [hunk 1]
```diff
@@ -26,7 +26,7 @@ global
   daemon
 <% end -%>
   <%= @global['debug_option'] %>
-<% @global['stats'].each do |option, value| -%>
+<% @global['stats']&.each do |option, value| -%>
```
Comment (damacus) [inline templates/default/haproxy.cfg.erb]: Excuse my Ruby/IRB ignorance here, what do this do?

**Comment by Stromweld** [comment]
ruby safe navigation operator. It removes the need to check for nil value before calling a method that would throw an exception for calling x method on nil value/nil class. I've found it to be a convenient way to not need if/unless statements for loops.

In this case if `@global['stats']` is nill the whole block is skipped.

https://docs.ruby-lang.org/en/2.4.0/NEWS-2_3_0.html

**Comment by damacus** [comment]
Neat. Thanks for the docs links!

**Pull request merged**